Ignore BaseURL option

mihaigeorgescu-newstore opened the pull request:
This PR adds the possibility to ignore the parts of the url when comparing requests.
I added `ignoreURLComponents` options to the `Session`. This is useful when you are using two environments, sandbox and production. And the request may have a different scheme, port or host.

mihaigeorgescu-newstore committed:
Added possibility to ignoreBaseURL when comparing request url with cassette url.

mihaigeorgescu-newstore committed:
Add documentation for ignoreBaseURL

mihaigeorgescu-newstore commented:
So far nobody is interested in this feature, how do you handle this situations when you have different URL for Sandbox and Production ?

soffes commented:
I am very interested in this. I've been busy though. Maybe @hyperspacemark has time to review?

soffes commented:
Thanks for your work on this @mihaigeorgescu-newstore!

eliperkins commented on DVR/Cassette.swift:
```diff
@@ -53,6 +57,31 @@ extension Cassette {
+        if ignoreBaseURL {
+            if let firstRelativePath = self.relativePath,
+                secondRelativePath = url.relativePath {
+                    isEqualWithURL = firstRelativePath == secondRelativePath
```
Hm. What about things like query parameters or fragments? Those seem important to check as well, for equality-sake.

mihaigeorgescu-newstore replied:
I will redo the check to take in consideration other components besides relative path, and create some unit tests it.

eliperkins commented on DVR/Session.swift:
```diff
@@ -34,7 +39,7 @@ public class Session: NSURLSession {
         return addDataTask(request)
     }
-
+    
```
✂️ 

Can you revert this whitespace change?

eliperkins commented:
📫

mihaigeorgescu-newstore commented:
PR was updated, the url check is performed directly on the URLComponents. @eliperkins could you have another look at it ?

Commit:
Replace print() + abort() with fatalError() so Quick Testing Framework correctly handles error

Commit:
Convert DVR to Swift3

Commit:
wip

newmarcel committed:
Xcode 8.0 (Swift 3.0.0) support

Commit:
PR Feedback

Commit:
fix-upload-tests

Commit:
Merge pull request #2 from NewStore/swift3

NA-9167: Swift3 version of DVR

eliperkins committed:
Swift 3 support (#65)

* Update to Swift 3

* Bump version 0.4.0

* Rename HTTPURLResponse file too

* Prefer private over fileprivate

* Fix serializating HTTPURLResponse

* Don't say we failed if we didn't

* Swift 3.0.1

* tvOS target

* Swift 3.0.1

* Fix tvOS deployment target

* Fix upload tests

* Explicitly set SWIFT_VERSION

Without this, Xcode complains about the legacy Swift version not being
set.

* Remove explicit toolchain from CI

* Specify SDK for CI

* Add destinations for CI

...because Travis CI is so good.

* Use latest version of Xcode

* Clean up spacing

hyperspacemark committed:
Add Swift 3 version compatibility

hyperspacemark committed:
Remove redundant tuple

Resolves a build warning in Xcode 9

hyperspacemark committed:
Deserialize interaction date as TimeInterval

This fails to deserialize as an Int in Swift 3.2.

soffes committed:
Update podspec to 1.0.1

Fixes #66

florianbuerger committed:
Expose tvOS support

Commit:
Fixed a conversion error

Commit:
Raised version to 0.4.8n2

hyperspacemark committed:
Update to Xcode 9 recommended build settings

hyperspacemark committed:
Update to Xcode 9

hyperspacemark committed:
Migrate to Swift 4

hyperspacemark committed:
Remove invalid override in extension

Swift 4 complains that properties and methods cannot be overridden in
extensions. I’m not sure if this ever worked before but regardless it’s
now invalid and needs to be removed.

Instead of using the override to hopefully get this behavior when
calling `dictionary` on an `HTTPURLResponse`, we’ll just check to see
if the response is an `HTTPURLResponse` and add the appropriate values
when converting the response into a dictionary in
`Interaction.dictionary`.

hyperspacemark committed:
Bump version to 1.1.0

Commit:
Change directory layout to support Swift Package Manager

Commit:
Add Package.swift; update .gitignore; update .swift-version file

Commit:
Add missing import statements to enable Swift build

Commit:
Update .travis.yml for Xcode 9 and add seperate build script

Add swift build to the script

Commit:
Add SPM badge to readme

Commit:
Use SPM badge that is more consistent with other badges

Commit:
Fixes HTTP status codes not recorded when not 200. #74

Commit:
Updated for Xcode 9

Commit:
updating travis osx_image to 9

soffes committed:
Update readme for Swift 4

hyperspacemark committed:
Update Travis builds to Xcode 9

soffes committed:
Update Travis macOS target name

Commit:
Raised version to 0.4.8n4

Commit:
Updated for Xcode 9.3

Commit:
Updated for Xcode 9.4

loudmouth committed:
Debug status code not being recorded

loudmouth committed:
Update xcode settings and to Swift 4.1

loudmouth committed:
Use travis' built-in xcpretty

loudmouth committed:
Update file path creation to use FileManager in tests

loudmouth committed:
Revert travis destination for OS X

eliperkins committed:
Bump podspec to version 1.2.0

loudmouth committed:
Add travis job for tvOS

hyperspacemark committed:
Replace open let with public let

Open let doesn't make sense, constants are implicitly final.

hyperspacemark committed:
Fix Swift 4.1 compiler error

"instance method 'filter' is internal and cannot be referenced from a default argument value"

Extracted the default test bundle into a public static Bundle? var while continuing to force unwrap it as the default argument.

hyperspacemark committed:
Fix build error

The iOS framework target was copy the test target's info plist into the bundle. New build system caught it.

hyperspacemark committed:
Remove duplicate file references

Commit:
Updated for Xcode 10

Commit:
Version 0.4.8n8

Commit:
Bump all targets to Swift Version 4.2

Commit:
Readme Build Status

olegdanu-newstore committed:
Added possibility to ignoreBaseURL when comparing request url with cassette url.

olegdanu-newstore committed:
Swift 4.2 support #6

Swift 4.2 support #6

olegdanu-newstore committed:
Version 1.3.1

Commit:
Version 1.3.1n2

eliperkins commented:
Closing as these branches have diverged too far for this PR to be valuable
